src/Features/Core/Portable/CodeFixes/AddImport/AbstractAddImportCodeFixProvider.cs

```diff
@@ -265,6 +267,42 @@ internal abstract partial class AbstractAddImportCodeFixProvider<TSimpleNameSynt
         }
 
         /// <summary>
+        /// We ignore references that are in a directory that contains the names "Packages".
+        /// These directories are most likely the ones produced by NuGet, and we don't want
+        /// to offer to add .dll reference manually for dlls that are part of NuGet packages.
+        /// </summary>
+        private bool IsInPackagesDirectory(PortableExecutableReference reference)
+        {
+            const string Packages = nameof(Packages);
+
+            try
+            {
+                if (!string.IsNullOrWhiteSpace(reference.FilePath))
+                {
+                    if (reference.FilePath.IndexOf(Packages, StringComparison.OrdinalIgnoreCase) >= 0)
```

Comment:
> No.

I don't get the concern.  This is a reasonable heuristic we're implementing that will do the right thing in practically all cases for all our customers.  It prevents the very unpleasant experience of us offering to add a direct metadata reference to something that should only be referenced as a nuget package.

It does mean that *if* the following is true:

1.  You have a project that has a non-nuget metadata reference to something in a "packages" directory, and
2. You are in another project that uses a type name that would have matched an accessible type from that dll.

then we will not offer to add that .dll reference to that other project.

That's it.  That's the only downside.  So we are slightly degraded in an exceedingly uncommon case, and we're vastly improved in the common case.  This is a totally acceptable and desirable outcome for such a heuristic.